unify seq rw input_dist

YazhiGao opened the pull request:
Summary:
* unify unnecessary branching for input_dist module
* fx wrap some splits for honoring non-optional points.

Differential Revision: D57876357

jingsh reviewed (approved):
can you fix the pyre, it's an easy fix

Commit:
unify seq rw input_dist (#2051)

Summary:

* unify unnecessary branching for input_dist module
* fx wrap some splits for honoring non-optional points.

Reviewed By: jingsh, gnahzg, yumin829928

Differential Revision: D57876357
